### ISSUE TITLE
TaskProperty needs to be updated to be a stage 1 decorator

### ISSUE THREAD

**Issue by pzuraq** [ember-animation/ember-animated#75]
Heya, so we merged some major refactors upstream to Ember's Descriptor system, we thought we'd found all users of it but missed this library. The gist now is that Descriptors are now stage 1 decorators, and Ember.defineProperty knows how to apply them as such.

You can see how we updated [Ember Concurrency here](https://github.com/machty/ember-concurrency/pull/277), and using `ember-compatibility-helpers` it should be possible to do something similar here without breaking changes. I'll try to submit a PR as soon as possible, but if anyone wants to take this on in the meantime let me know, I'm happy to help and provide guidance!

**Comment by geoffreyd** [comment]
@pzuraq I've taken a shot, it'd be great to get some eyes to make sure I took the right approach.

**Comment by ef4** [comment]
Should be fixed in 0.5.1.

**Comment by MichalBryxi** [comment]
Just in case someone needs to google for this: *Ember octane preview* app with `ember-animated v0.5.0` caused for me:

```
index.js:163 Uncaught (in promise) Error: Assertion Failed: A computed property key must be a string, you passed undefined undefined
    at assert (index.js:163)
    at expandProperties (metal.js:2507)
    at TaskProperty._property (metal.js:2987)
    at TaskProperty.ComputedProperty (metal.js:2845)
    at new TaskProperty (ember-scheduler.js:34)
    at _construct (construct.js:36)
    at task (ember-scheduler.js:19)
    at Module.callback (animated-each.js:535)
    at Module.exports (loader.js:106)
    at Module._reify (loader.js:143)
```

When using:

```
<AnimatedContainer></AnimatedContainer>
```

possibly even other components from this library. Upgrading to `ember-animated v0.5.1` solved that issue for me.
